Update miden-node.md > node design diagram

gubloon opened the pull request:
Trying to improve the node design diagram

bobbinth reviewed (commented):
Thank you! A few thoughts:

1. I would probably not change the style of the diagram as we use similar style for most other diagrams and it would be good to have one consistent style throughout (the tool we use to create diagrams is [Excalidraw](https://app.excalidraw.com/).
2. I'm not sure I like the flow from top to bottom. The flow we have right now (right to left) is not great either though. Maybe we should try to do left-to-right?
3. In the RPC, `sync state`, `get_block_header`, and `check_nullifiers` are the `data retrieval endpoints`. So, if we have them as separate boxes, we don't need to have `data retrieval endpoints`.
4. A line from block builder to state view is missing. But also, more generally, the lines are a bit more difficult to follow than in the original (though, I like adding more label to lines).

bobbinth commented:
@gubloon - are you planning to update the diagram - or should we close this PR?

gubloon commented:
> @gubloon - are you planning to update the diagram - or should we close this PR?

I had ~2 version based on your input and wanted to know your feedback on them; I will update them by today @bobbinth

gubloon commented:
Thanks for the feedback Bobbin; here's an updated diagram:
![Miden Node architecture](https://github.com/0xPolygonMiden/miden-base/assets/7201911/688cb7cb-157d-4c19-8539-5e1267a6446f)

1. Reg. the style, I did the previous diagram in Excalidraw too but tried the style where the lines are not sloppy and the font was `normal` rather than the `hand-drawn` one to see if that can contribute to making the diagram somewhat prettier and quickly readable as the rest of the docs also will have normal font and normal lines. I also had in mind that, if we have to change style for this one, we will have to do that for all the other diagrams too. 
However in this updated I reverted the style back to the moderate sloppiness with hand-drawn font as we don't want to change the style. 
2. Yes, in this diagram I updated the flow to be only from left to right.
3. Updated likewise. 
4. Fixed and tried to make lines more consistent and to be easy to follow. Also added labels to the lines - but not sure if that only adds more noise with additional boxes(we can keep them if you don't feel so). For the lines where the data is being read, I made the arrows double headed so that it represents both the flow of requests being sent to the DB and also responses coming from the DB(will be that be correct?)

gubloon commented:
Also, here is the same diagram with non-sloppy lines and normal font, just for a thought how the diagram will look in the other way:
![Miden Node architecture normal](https://github.com/0xPolygonMiden/miden-base/assets/7201911/76608a6f-29de-43bb-b5a6-0e42151d5ce2)

Dominik1999 commented:
Should we merge or close this?

gubloon commented:
If the diagram [here](https://github.com/0xPolygonMiden/miden-base/pull/459#issuecomment-1978173784) is worth the change, I can update it and we can merge. We can simply close this PR, if otherwise.

Dominik1999 commented:
ok we want to use that, but can you 

- reduce size of tx validator box smaller
- somehow can you make the block builder bigger
- we now have more endpoints (get account details, get public notes - please check the real names with @phklive )
- can you use the first version (hand drawn excalidraw)

gubloon commented:
> ok we want to use that, but can you
> 
> * reduce size of tx validator box smaller
> * somehow can you make the block builder bigger
> * we now have more endpoints (get account details, get public notes - please check the real names with @phklive )
> * can you use the first version (hand drawn excalidraw)

@Dominik1999 I've updated the diagram as per above input:
![Miden Node](https://github.com/0xPolygonMiden/miden-base/assets/7201911/6fec020a-8071-472b-ad43-38324190038f)

bobbinth commented:
Looks good! Thank you! A couple more small suggestions:

1. The arrow from `tx validator` to `mempool` should actually connect to the `proven tx queue` (not the `tx batch queue`).
2. I would move `tx validator`, `tx compiler`, and `tx executor` boxes down, and would stretch `state view` to the left so that it is above both `tx validator` and `block builder`.
3. I would move `SyncState` down to be in the same group as other endpoints. I would list endpoints in the following order:
  a. `GetBlockHeaderByNumber`
  b. `GetAccountDetails`
  c. `GetNotesById`
  d. `SyncState`
  e. `CheckNullifiers`

gubloon commented:
> Looks good! Thank you! A couple more small suggestions:
> 
> 1. The arrow from `tx validator` to `mempool` should actually connect to the `proven tx queue` (not the `tx batch queue`).
> 2. I would move `tx validator`, `tx compiler`, and `tx executor` boxes down, and would stretch `state view` to the left so that it is above both `tx validator` and `block builder`.
> 3. I would move `SyncState` down to be in the same group as other endpoints. I would list endpoints in the following order:
>    a. `GetBlockHeaderByNumber`
>    b. `GetAccountDetails`
>    c. `GetNotesById`
>    d. `SyncState`
>    e. `CheckNullifiers`

Thanks @bobbinth; I've update the diagram as per above feedback:
![Miden Node](https://github.com/0xPolygonMiden/miden-base/assets/7201911/0f5a6d8b-9723-40f9-8b4d-755eb94521fc)

gubloon commented:
> Looks good! Thank you! A couple more small suggestions:
> 
> 1. The arrow from `tx validator` to `mempool` should actually connect to the `proven tx queue` (not the `tx batch queue`).
> 2. I would move `tx validator`, `tx compiler`, and `tx executor` boxes down, and would stretch `state view` to the left so that it is above both `tx validator` and `block builder`.
> 3. I would move `SyncState` down to be in the same group as other endpoints. I would list endpoints in the following order:
>    a. `GetBlockHeaderByNumber`
>    b. `GetAccountDetails`
>    c. `GetNotesById`
>    d. `SyncState`
>    e. `CheckNullifiers`

Thanks @bobbinth; I've updated the diagram as per above feedback:
![Miden Node](https://github.com/0xPolygonMiden/miden-base/assets/7201911/f0de5bfb-284a-4584-8b16-6775a58782b3)

bobbinth commented:
Looks great! Thank you again! There seems to be a conflict in this PR. Could you resolve the conflict (or maybe just open another PR against the latest `next` branch)?
